### PR TITLE
Upgrade to Symfony 3.1

### PR DESCRIPTION
**Pull request by ahilles107**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | yes
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | 
| License                   | AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/ContentBundle/spec/Form/Type/RouteSelectorTypeSpec.php [hunk 1]
```diff
@@ -28,22 +28,22 @@
...
 final class RouteSelectorTypeSpec extends ObjectBehavior
 {
-    function let(RouteProviderInterface $routeProvider)
+    public function let(RouteProviderInterface $routeProvider)
     {
         $this->beConstructedWith($routeProvider);
     }
 
-    function it_is_initializable()
+    public function it_is_initializable()
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/spec/Form/Type/RouteSelectorTypeSpec.php]: methods here don't need to be pubic here

File: src/SWP/Bundle/CoreBundle/DependencyInjection/Compiler/OverrideDynamicRouterPass.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * @copyright 2016 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use SWP\Bundle\CoreBundle\Routing\MetaRouter;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class OverrideDynamicRouterPass extends AbstractOverridePass
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/DependencyInjection/Compiler/OverrideDynamicRouterPass.php]: missing tests

File: src/SWP/Bundle/CoreBundle/Version/Version.php [hunk 3]
```diff
@@ -14,12 +14,12 @@
 
 namespace SWP\Bundle\CoreBundle\Version;
 
-use SWP\UpdaterBundle\Version\VersionInterface;
+//use SWP\UpdaterBundle\Version\VersionInterface;
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/Version/Version.php]: can be removed for now

File: src/SWP/Bundle/RuleBundle/spec/Form/Type/RuleTypeSpec.php [hunk 4]
```diff
@@ -28,17 +29,17 @@
  */
 final class RuleTypeSpec extends ObjectBehavior
 {
-    function it_is_initializable()
+    public function it_is_initializable()
```
Comment (takeit) [inline src/SWP/Bundle/RuleBundle/spec/Form/Type/RuleTypeSpec.php]: no need to explicitly specifying public scope in this spec

Comment (ahilles107) [inline src/SWP/Bundle/RuleBundle/spec/Form/Type/RuleTypeSpec.php]: public is by default and it's good practice to define it even if it's default type. And it was fixed automaticaly by my phpcs fixer.

**Pull request merged**